### PR TITLE
Packaging, testing, & syntax improvements

### PR DESCRIPTION
**Pull request by bsolomon1124**
Hey @scottfrazer , nice to meet you.  I'm very glad to have found this package and have some additions here that I think will make it more robust.

Changes & additions:

- More comprehensive testing: test via tox + pytest across 2.7 thru 3.7
- Docstring/PEP8/linted
- More performant & efficient
- More defined in terms of the rules that the parser accepts.  Specifically, take additional cues from [GitHub-Flavored Markdown](https://github.github.com/gfm/#atx-heading), because other older specifications leave a lot to be guessed.

I know it's a fairly large changeset, so please take look at the diff and let me know your thoughts.

### CODE REVIEW

**Comment by scottfrazer** [comment]
Cool, much better :)

**Review by scottfrazer — APPROVED** [review]

**Comment by bsolomon1124** [comment]
Awesome @scottfrazer .  One more small suggestion: upgrade the `__version__` (just internal changes, no big API changes here) and push up to PyPI?

File: mdtoc/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+__version__ = "0.1"
+__author__ = "Scott Frazer <scott.d.frazer@gmail.com>"
```
Comment (scottfrazer) [inline mdtoc/__init__.py:2]: @bsolomon1124 This is not quite accurate anymore!  you should be the author or at least co-author

**Comment by scottfrazer** [comment]
Yeah I can do that... I'll get to that soonish

File: mdtoc/__init__.py (hunk 1 above)
Comment (bsolomon1124) [reply mdtoc/__init__.py:2]: @scottfrazer How's this? https://github.com/scottfrazer/mdtoc/pull/2 😁

**Pull request merged**